admiral/pkg/controller/admiral/node.go

```diff
@@ -67,7 +68,7 @@ func process(ctx context.Context, obj interface{}) (string, error) {
 
 func (d *NodeController) Added(ctx context.Context, obj interface{}) error {
 	region, err := process(ctx, obj)
-	if err != nil {
+	if err != nil || region == "" {
```

Comment:
can we have separate if with new error message

Reply:
+1

Reply:
@Punakshi can you add a test for this?

Reply:
Added